src/Widgets/calendar/GridDay.vala

```diff
@@ -70,20 +69,70 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
 
         label.name = "date";
 
-        add (label);
+        main_grid = new Gtk.Grid ();
+        main_grid.hexpand = true;
+        main_grid.halign = Gtk.Align.CENTER;
+        main_grid.valign = Gtk.Align.CENTER;
+        main_grid.orientation = Gtk.Orientation.VERTICAL;
+        main_grid.attach (label, 0, 0);
+
+        add (main_grid);
+        halign = Gtk.Align.CENTER;
         show_all ();
 
         // Signals and handlers
         button_press_event.connect (on_button_press);
         key_press_event.connect (on_key_press);
         scroll_event.connect ((event) => {return Util.on_scroll_event (event);});
+
+        Widgets.CalendarModel.get_default ().events_added.connect (update_event_days);
```

Comment:
You should probably cache this so you're not getting the calendar model three times in a row